docs/01-Using-Fleet/13-Vulnerability-Processing.md

```diff
@@ -83,3 +84,115 @@ The matching occurs server-side to make the processing as fast as possible, but
 For example, when running a development instance of Fleet on an Apple Macbook Pro with 16 cores, matching 200,000 CPEs against the CVE database will take around 10 seconds and consume about 3GBs of RAM. 
 
 The CPU and memory usages are in burst once every hour (or the configured periodicity) on the instance that does the processing. RAM spikes are expected to not exceed the 2GBs.
+
+## Detection pipeline
+
+There are several steps that go into the vulnerability detection process. In this section we'll dive into what they are and how it works.
+
+The process has different parts that are more error-prone than others. Each OS and each application developer and maintainer can (and do) have their own way of defining each part of their app. Some Linux distributions are very strict, but each distribution handles things differently.
+
+The whole pipeline exists to compensate for these differences, and it can be divided in two sections:
+
+1. Collection:
+
+```mermaid
+graph TD;
+    A[Host1 send software list]-->D[Normalization of names, versions, etc]
+    B[Host2 send software list]-->D
+    C[Host3 send software list]-->D
+    D-->E[Storage for later processing]
+```
+
+2. Processing
+
+Processing happens in a loop. The default interval is 1hr.
+
+```mermaid
+graph TD;
+    A{Once an hour}-->B[Normalized software list]
+    B-->C[CPE translation]
+    C-->D[CVE detection]
+    D-->E[False positive removal]
+    E-->A
+```
```

Comment:
Two more steps to add:
- Before CPE translation, the CPE database is downloaded from the fleetdm/nvd github repo.
- Before CVE detection, the CVE information is synced from NVD site.